### PR TITLE
ranger: BuildColumnRange should merge ranges when column has prefix len

### PR DESCRIPTION
**Pull request by winoros**
### What problem does this PR solve? 

Fix the second case of #11544 

### What is changed and how it works?

`BuildColumnRange` should union the ranges if it has been cut.

### Check List 

Tests 

 - Unit test

Related changes

 - Need to cherry-pick to the release branch
 - Need to be included in the release note

### CODE REVIEW

File: executor/join_test.go [hunk 1]
```diff
@@ -1381,3 +1381,13 @@ func (s *testSuite2) TestInjectProjOnTopN(c *C) {
...
 }
+
+func (s *testSuite2) TestIssue11544(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec("use test")
+	tk.MustExec("create table 11544t(a int)")
+	tk.MustExec("create table 11544tt(a int, b varchar(10), index idx(a, b(3)))")
+	tk.MustExec("insert into 11544t values(1)")
+	tk.MustExec("insert into 11544tt values(1, 'aaaaaaa'), (1, 'aaaabbb'), (1, 'aaaacccc')")
+	tk.MustQuery("select /*+ TIDB_INLJ(11544tt) */ * from 11544t t, 11544tt tt where t.a=tt.a and (tt.b = 'aaaaaaa' or tt.b = 'aaaabbb')").Check(testkit.Rows("1 1 aaaaaaa", "1 1 aaaabbb"))
```
Comment (lamxTyler) [inline executor/join_test.go]: `There are no matching table names for (11544tt) in optimizer hint /*+ TIDB_INLJ(11544tt) */. Maybe you can use the table alias name`

Comment (winoros) [reply executor/join_test.go]: fixed

File: util/ranger/ranger.go [hunk 2]
```diff
@@ -270,6 +270,10 @@ func buildColumnRange(accessConditions []expression.Expression, sc *stmtctx.Stat
 				ran.HighExclude = false
 			}
 		}
+		ranges, err = unionRanges(sc, ranges)
```
Comment (lzmhhh123) [inline util/ranger/ranger.go]: Only if `colLen != types.UnspecifiedLength` should union the ranges?

Comment (winoros) [reply util/ranger/ranger.go]: Yes, only cuting by prefix will cause the case the ranges are overflapped.
`a in ('aaa', 'aab', 'aac')`
If the len is 1. The ranges after cutting before unioning is `[a, a], [a, a], [a, a]`.

**Pull request merged**